src/coord/dist_aggregate.c

```diff
@@ -228,11 +228,29 @@ void printAggProfile(RedisModule_Reply *reply, void *ctx) {
   // profileRP replace netRP as end PR
   ProfilePrinterCtx *cCtx = ctx;
   RPNet *rpnet = (RPNet *)AREQ_QueryProcessingCtx(cCtx->req)->rootProc;
+
+  // Therefore, we loop to drain all remaining replies from the channel.
+  // Pending might be zero, but there might still be replies in the channel to read.
+  // We may have pulled all the replies from the channel and arrived here due to a timeout,
+  // and now we're waiting for the profile results.
+  while (MRIterator_GetPending(rpnet->it) || MRIterator_GetChannelSize(rpnet->it)) {
+    getNextReply(rpnet);
+  }
+
+  size_t num_shards = GetNumShards_UnSafe();  // can we make this safe
```

Comment:
Can we get this from the iterator? It will also make more sense to take the number of commands we sent and not the current number of shards

Reply:
you mean it->len?